### DISCUSSION TITLE
Mapping Ctrl+w to close an editor tab seems impossible.

### DISCUSSION THREAD

**Discussion by Toerktumlare** [JetBrains/ideavim#369] [General]
I have read that Ctrl+Shift+F4 is mapped to `CloseActiveTab` but this does nothing.

After googling some more jetbrains seem to have decided to make the action `CloseActiveEditor` the action to run to close the active tab im working in.

I want to be able to close the active tab by using Ctrl+w, a very simple task.

```
nnoremap <C-w> <Action>(CloseActiveEditor)
```

But close active editor does not exist in the action list, but you can find it in the keymappings för the regular idea.

And <C-S-F4> does nothing. And mapping to action `CloseActiveTab` does nothing.

How can doing such a simple task be so difficult?

**Comment by AlexPl292** [comment]
Hi! Could you please try `<Action>(CloseContent)`?
I have found it using `:actionlist close`, but huh, there are a lot of unrelated actions.

**Reply by Toerktumlare** [reply]
thank you for your reply.

using the command

```
map <C-w> <Action>(CloseContent)
```

does seem to work, but this gives me more questions.

why does `map` work, but not `nnoremap`?

if i run it using `nnoremap` it will execute my command as `(` and `C` and then insert `loseContent)` after. This must be a bug?

tbh, i find the docs to ideavim very confusing, and there should be a lot more provided examples and explanations of standard expected tasks. No there is just a massive actionlist and no explanation of any of them.

**Comment by AlexPl292** [comment]
Glad to hear that everything works now!

> why does map work, but not nnoremap?

This is actually not a bug, but a bad design decision. This approach is more correct from a technical point of view (for example, you also can't create a no-remap mapping to plugin commands `<Plug>`), but now it's clear that this confuses people a lot and gives no value. We want to change it to support no-remap for `<Action>`.

> i find the docs to ideavim very confusing

Could you please elaborate a little bit about it? Do you mean github readme docs or the result of `:actionlist` command? Do you mean the action execution, or the whole docs?
Actually, we rewrite and renew docs from time to time to keep them clear and updated, so maybe it's time to perform a cleanup loop again.

**Reply by Toerktumlare** [reply]
Thank you for your response.

I'd say a bit of both. Im a decent vim user, and started quite recently using ideavim also for when i do java development in intellij.

And what struck me first then it came to the documentation, was that there was no clear list of what commands where supported from vim into idea. There has been a bit of trial and error. And for instance the map, vs noremap, inoremap imap nmap etc etc, when does what work.

And the actionlist here [ActionList](https://github.com/JetBrains/ideavim/blob/master/src/com/maddyhome/idea/vim/package-info.java) mostly just links to a class. But what the action actually does, is once again, trial and error.

Id like to se a more extensive file example, showing a bigger flora of both setting vim features, but also displaying a bunch of example keybindings to some common actions. with some explanation of what the action does.

For instance, mapping Ctrl-w to close a window, which must be a pretty common command.

But i dont know, i might be nitpicking, i just find it much easier to learn by seeing examples, than reading a list of commands.

**Reply by citizenmatt** [reply]
I'd vote for keeping the map behaviour of `<Action>(…)`, as it mirrors how `<Plug>` works. I'd also argue that it's logical (if not exactly intuitive) because it needs to map that text to the action. If `nomap` calls the action, how would you create a mapping to insert the text `<Action>`? It might be a somewhat contrived example, but it's a good reason for keeping the map.

**Reply by AlexPl292** [reply]
This is a fair point, but the problem is the this is not the first case when `noremap` causes trouble with `<Action>`. Well, maybe we can put a big yellow label in docs that `<Action>` should be used only with `map` and see if this will fix the problem.

**Reply by Toerktumlare** [reply]
im not against one or the other, its just not intuitive, so as pointed out, if the docs at least points this out (what seems to be a common) misconception its fine. Just writing the explanation to why its designed the way it is and how one should go about is enough in the readme. Because currently it does not mention anything of it, so i'd say it's natural that people will use <Action> with noremap.

**Reply by lippfi** [reply]
I've never used `<Plug>` mappings in my .vimrc and didn't know about this Vim behaviour until today, so at first I thought that it is a bug in IdeaVim.
If we aren't going to change the default behaviour for noremap with `<Action>`, then we should at least somehow notify user that potentially unwanted mapping is used
